### DISCUSSION TITLE
Is there any way to have different error messages for different required errors?

### DISCUSSION THREAD

**Discussion by philippeadam** [solspace/craft-freeform#1541] [Feature Requests]
I would like to have different required messages for different errors like so:
- Your fist name is required
- Your last name is required

I would also like to keep using the craft.freeform.form.render() and avoid rendering the form manually.

Thanks

**Comment by kjmartens** [comment]
Sorry for the delay @philippeadam,

Unfortunately this is currently not possible. However, it's on our list for future consideration. 🙂

**Comment by philippeadam** [comment]
@kjmartens Ok I really think this should be done in the future as it is currently a fail of the WCAG criteria :

3.3.1 Error Identification - Level A

**Reply by kjmartens** [reply]
I discussed this with our lead developer and we will add support for it within the next week or two. 🙂

**Comment by kjmartens** [comment]
This is now available in Freeform 5.7.0 🙂
